models/Asset/Image/Thumbnail/Processor.php

```diff
@@ -178,21 +179,37 @@ public static function process(Asset $asset, Config $config, $fileSystemPath = n
         $storage = Storage::get('thumbnail');
 
         // check for existing and still valid thumbnail
-        if ($storage->fileExists($storagePath)) {
-            if ($storage->lastModified($storagePath) >= $asset->getModificationDate()) {
-                return [
-                    'src' => $storagePath,
-                    'type' => 'thumbnail',
-                    'storagePath' => $storagePath,
-                ];
-            } else {
-                // delete the file if it's not valid anymore, otherwise writing the actual data from
-                // the local tmp-file to the real storage a bit further down doesn't work, as it has a
-                // check for race-conditions & locking, so it needs to check for the existence of the thumbnail
-                $storage->delete($storagePath);
+
+        $modificationDate = null;
+        $statusCacheEnabled = \Pimcore::getContainer()->getParameter('pimcore.config')['assets']['image']['thumbnails']['status_cache'];
+        if($statusCacheEnabled) {
+            $modificationDate = $asset->getDao()->hasThumbnail($config->getName(), $filename);
```

Comment:
add additional fallback in that case the thumbnail is not in cache ... but might exist on file system.

Reply:
Doing so would make the cache obsolete ... the whole point of this cache is to avoid these checks by assuming that the thumbnail doesn't exist if it's not in the cache. 

Consider the following example: 
You have a new image, and you request `$thumb->getHtml()` for it. For that you need to generate 7-8 thumbnails for the `<picture>` tag. Without having the fallback there wouldn't be a single request to the filesystem, because the assumtion is that the thumbnail is not generated when not being in cache. So if we do implement a fallback, then you'd have 7-8 requests to the filesystem, because there's nothing in the cache because no thumb was generated.